pennylane/tape/tapes/cv_param_shift.py

```diff
@@ -456,11 +489,35 @@ def parameter_shift_var(self, idx, device, params, **options):
             # with itself, to get a square symmetric matrix representing
             # the square of the observable
             obs = qml.PolyXP(np.outer(A, A), wires=obs.wires, do_queue=False)
-            temp_tape._measurements[i] = MeasurementProcess(qml.operation.Expectation, obs=obs)
+            pdA2_tape._measurements[i] = MeasurementProcess(qml.operation.Expectation, obs=obs)
 
         # Here, we calculate the analytic derivatives of the <A^2> observables.
-        pdA2 = temp_tape.parameter_shift_second_order(idx, device, params, **options)
+        pdA2_tapes, pdA2_fn = pdA2_tape.parameter_shift_second_order(idx, params, **options)
+        tapes.extend(pdA2_tapes)
+
+        if self._evA is None:
+            tapes.append(evA_tape)
+
+        def processing_fn(results):
+            """Computes the gradient of the parameter at index ``idx`` via the
+            second order CV parameter-shift method for a circuit containing a mixture
+            of expectation values and variances.
+
+            Args:
+                results (list[real]): evaluated quantum tapes
+
+            Returns:
+                array[float]: 1-dimensional array of length determined by the tape output
+                measurement statistics
+            """
+            pdA = pdA_fn(results[0:2])
+            pdA2 = pdA2_fn(results[2:4])
+
+            if self._evA_result is None:
+                self._evA_result = np.array(results[-1])
+
+            # return d(var(A))/dp = d<A^2>/dp -2 * <A> * d<A>/dp for the variances,
+            # d<A>/dp for plain expectations
+            return np.where(self.var_mask, pdA2 - 2 * self._evA_result * pdA, pdA)
 
-        # return d(var(A))/dp = d<A^2>/dp -2 * <A> * d<A>/dp for the variances,
-        # d<A>/dp for plain expectations
-        return np.where(self.var_mask, pdA2 - 2 * self._evA * pdA, pdA)
+        return tapes, processing_fn

```

Comment:
This code seems to entirely mirror the same code for qubit param shift. Since it's already hard to parse, maybe we could move this logic into a separate shared location in the codebase?

Reply:
I would argue no; while the code appears identical comparing them line by line, they aren't conceptually identical. Improvements I can see that we should include in the CV version long-term will eventually result in the two methods diverging. I think this is a case of [prefering duplication over the wrong abstraction](https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction?duplication).

Reply:
Instead, I would suggest making both the qubit and CV `parameter_shift_var` methods better commented, documented, and easier to read.